### PR TITLE
feat: add disable apigateway auto create, close #12

### LINKED ISSUE #12: api配置不支持自定义域名，也不支持关闭自动创建api网关
希望能支持api网关的所有配置，或者提供不使用自动创建API网关的选项，让用户自行创建需要的API网关服务

### PR DESCRIPTION
**Pull request by yugasun**

### CODE REVIEW

**Comment by tinafangkunding** [comment]
@pmuens @medikoo could you please help to review and merge this pr? Thanks a lot!

**Review by tinafangkunding — DISMISSED** [review]

**Review by medikoo — APPROVED** [review]

File: serverless.js [hunk 1]
```diff
@@ -49,39 +48,48 @@ module.exports = class TencentKoa extends Component {
...
+      functionName: inputs.name,
     };
-    if (apigatewayConf.usagePlan) apigwParam.endpoints[0].usagePlan = apigatewayConf.usagePlan;
-    if (apigatewayConf.auth) apigwParam.endpoints[0].auth = inputs.apigatewayConf.auth;
 
-    this.state.functionName = inputs.name;
-    await this.save();
-    const tencentApiGatewayOutputs = await tencentApiGateway(apigwParam);
+    // if not disable, then create apigateway
+    if (apigatewayConf.disable !== true) {
```
Comment (medikoo) [inline serverless.js]: When handling booleans in JS, best working way is to simply distinguish between _truthy_ and _falsy_ values.

So cleanest approach would be:

```javascript
if (!apigatewayConf.isDisabled) {
...
```

File: docs/configure.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+| Param    | Description      |
+| -------- | :--------------- |
+| subnetId | ID of the VPC    |
+| vpcId    | ID of the subnet |
+
+### apigatewayConf param description
+
+| Param       | Required/Optional | Description                                                                                              |
+| ----------- | :---------------: | :------------------------------------------------------------------------------------------------------- |
+| disable     |     Optional      | Whether disable auto creating api gateway, if you don't want to create it, set to `true`                 |
```
Comment (medikoo) [inline docs/configure.md]: When naming boolean properties, it's good to prefix them with `is*`, `has*`. It gives clear indication it's a boolean.

In code `disable` sounds as method, through which we disable given object. So best name here would be `isDisabled`

**Review by medikoo — CHANGES_REQUESTED** [review]
Thanks @yugasun I've proposed few minor improvements

File: docs/configure.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+```yml
+# serverless.yml
+
+koa:
+  region: ap-shanghai
+  component: '@serverless/tencent-koa'
+  inputs:
+    region: ap-shanghai
+    functionName: eslam-function
```
Comment (pmuens) [inline docs/configure.md]: While it's only docs, could we make them more generic. So using e.g. `my-function` rather than `eslam-function`?

**Comment by yugasun** [comment]
@pmuens @medikoo Thanks for the reviewing~ I have updated `disable` to `isDisabled`, could you review again?

**Review by medikoo — APPROVED** [review]
Thank you @yugasun !

**Review by pmuens — APPROVED** [review]

**Pull request merged**